### DISCUSSION TITLE
Status of the rustdesk for sway

### DISCUSSION THREAD

**Discussion by Dzordzu** [rustdesk/rustdesk#7323] [Show and tell]
This discussion will try to summarize rustdesk status for the sway

### Absolute inability to connect to wayland host

As of now establishing connection with wayland host as the destination is impossible. It is strongly bound to the [missing  RemoteDesktop dbus service](https://github.com/emersion/xdg-desktop-portal-wlr/issues/2). Hopefully, it'll work after [this PR](https://github.com/emersion/xdg-desktop-portal-wlr/pull/263)

### Gentoo

Although currently it's impossible to connect to the wayland host, in the future it would eventually require some additional requirement. One must then install [`gui-libs/xdg-desktop-portal-wlr`](https://packages.gentoo.org/packages/gui-libs/xdg-desktop-portal-wlr)

### Archlinux

Similar situation as one with the gentoo - [`xdg-desktop-portal-wlr`](https://archlinux.org/packages/?name=xdg-desktop-portal-wlr)

**Comment by flowthentic** [comment]
I am starting my desktop session via uwsm, called by greetd.
I am able to connect, but usually get black screen. On very rare occasion I can share my screen. I can always connect to someone else's screen.

**Comment by miullu** [comment]
black screen also.
it say it is taking window 'null'

**Comment by rustdesk** [comment]
Does this https://github.com/rustdesk/rustdesk/discussions/8935#discussioncomment-13324800 work for you?

**Reply by miullu** [reply]
no

here is the whole log of rustdesk after installing that xdg-desktop-portal-wlr which is same as before
```rustdesk
Jun 07 21:01:50 archlinux rustdesk[184120]: flutter: Bring window 'null' on top
Jun 07 21:02:05 archlinux rustdesk[184120]: flutter: Bring window 'null' on top
Jun 07 21:02:19 archlinux rustdesk[184120]: flutter: model  closed
Jun 07 21:02:19 archlinux rustdesk[184120]: void fl_view_redraw(FlView *): assertion 'FL_IS_VIEW(self)' failed
Jun 07 21:02:19 archlinux rustdesk[184120]: gtk_window_get_position: assertion 'GTK_IS_WINDOW (window)' failed
Jun 07 21:02:19 archlinux rustdesk[184120]: gtk_window_get_size: assertion 'GTK_IS_WINDOW (window)' failed
Jun 07 21:02:19 archlinux rustdesk[184120]: gtk_widget_hide: assertion 'GTK_IS_WIDGET (widget)' failed
Jun 07 21:02:19 archlinux rustdesk[184120]: gtk_window_move: assertion 'GTK_IS_WINDOW (window)' failed
Jun 07 21:02:19 archlinux rustdesk[184120]: gtk_window_resize: assertion 'GTK_IS_WINDOW (window)' failed
Jun 07 21:02:19 archlinux rustdesk[184120]: gtk_widget_get_opacity: assertion 'GTK_IS_WIDGET (widget)' failed
```

**Reply by flowthentic** [reply]
My issue have been fixed by 1.3.9 release. Since then I can share my screen.

**Comment by itsvyle** [comment]
Is there any change on that matter?
